src/kernel/storage.c

```diff
@@ -263,7 +273,7 @@ closure_function(1, 2, boolean, volume_add_mount_each,
     return true;
 }
 
-boolean volume_add(u8 *uuid, char *label, storage_req_handler req_handler, u64 size, int attach_id)
+boolean volume_add(u8 *uuid, char *label, storage_req_handler req_handler, u64 size, int attach_id, boolean readonly)
```

Comment:
You're right, volume_mount is the right place for the read-only check to go. I've updated and cleaned it up here so that the checks are contained to only volume_mount().